plugins/modules/podman_generate_systemd.py

```diff
@@ -0,0 +1,525 @@
+  containers.podman.podman_container:
+    name: postgres_local
+    image: docker.io/library/postgres:latest
+    state: stopped
+
+- name: Systemd unit files for postgres container must exist
+  containers.podman.podman_generate_systemd:
+    name: postgres_local
+    path: ~/.config/systemd/user/
+
+- name: Postgres container must be started and enabled on systemd
+  ansible.builtin.systemd:
+    name: container-postgres_local
+    daemon_reload: yes
+    state: started
+    enabled: yes
+
+
+# Generate the unit files, but store them on an Ansible variable
+# instead of writting them on target host
+- name: Systemd unit files for postgres container must be generated
+  containers.podman.podman_generate_systemd:
+    name: postgres_local
+  register: postgres_local_systemd_unit
+
+# Generate the unit files with environment variables sets
+- name: Systemd unit files for postgres container must be generated
+  containers.podman.podman_generate_systemd:
+    name: postgres_local
+    env:
+      POSTGRES_USER: my_app
+      POSTGRES_PASSWORD: example
+  register: postgres_local_systemd_unit
+'''
+
+RETURN = '''
+systemd_units:
+  description: A copy of the generated systemd .service unit(s)
+  returned: always
+  type: dict
+  sample: {
+    "container-postgres_local": " #Content of the systemd .servec unit for postgres_local container",
+    "pod-my_webapp": " #Content of the systemd .servec unit for my_webapp pod",
+    }
+podman_command:
+  description: A copy of the podman command used to generate the systemd unit(s)
+  returned: always
+  type: str
+  sample: "podman generate systemd my_webapp"
+'''
+
+
+import os
+from ansible.module_utils.basic import AnsibleModule
+import json
+
+
+RESTART_POLICY_CHOICES = [
+    'no-restart',
+    'on-success',
+    'on-failure',
+    'on-abnormal',
+    'on-watchdog',
+    'on-abort',
+    'always',
+]
+
+
+def generate_systemd(module):
+    '''Generate systemd .service unit file from a pod or container.
+
+    Parameter:
+    - module (AnsibleModule): An AnsibleModule object
+
+    Returns (tuple[bool, list[str], str]):
+    - A boolean which indicate whether the targeted systemd state is modified
+    - A copy of the generated systemd .service units content
+    - A copy of the command, as a string
+    '''
+    # Flag which indicate whether the targeted system state is modified
+    changed = False
+
+    # Build the podman command, based on the module parameters
+    command_options = []
+
+    #  New option
+    if module.params.get('new'):
+        command_options.append('--new')
+
+    #  Restart policy option
+    restart_policy = module.params.get('restart_policy')
+    if restart_policy:
+        # If the restart policy requested is not supported by podman
+        if restart_policy not in RESTART_POLICY_CHOICES:
+            # Then stop the module execution and return an error message
+            module.fail_json(
+                msg=f'Restart policy requested is "{restart_policy}"'
```

Comment:
`ERROR: plugins/modules/podman_generate_systemd.py:253:69: traceback: SyntaxError: invalid syntax`
Can we please make it Py2 compatible?

Reply:
Do we target Python 2 ? I thought that we targeted Python 3.6.8. [from](https://github.com/containers/ansible-podman-collections/issues/491#issuecomment-1287773280)

Reply:
Sorry, probably I forgot by myself :laughing: But it's in todo list to drop Py2 support, since it requires to release the major collection version (2.0.0), which will be included in next major Ansible release only (because it's not backward compatible). So I'd like to collect all non-backward-compatible changes and release it later.

Reply:
Do I made modification to be Python 2 compatible ?

Reply:
Yeah, just use `.format` or `%s`. Seems like it's only the place?

Reply:
Would be great if you include this code as well: https://github.com/containers/ansible-podman-collections/pull/501 just copy paste it to your patch. That way we'll run CI jobs here.

Reply:
> Yeah, just use `.format` or `%s`. Seems like it's only the place?

I use a lot of f'' string.

Reply:
> Would be great if you include this code as well: #501 just copy paste it to your patch. That way we'll run CI jobs here.

Sorry, I did not understand what you meant.

Reply:
Code in https://github.com/containers/ansible-podman-collections/pull/501 is a skeleton for tests to run them in CI. Can you please take this code to your current patch?

Reply:
Done.

Reply:
Finally, what do I do about Python 2 ?

Reply:
We need to convert all F-strings to `format`. Let me know if you need help with that, I can do it.

Reply:
It's ok, I've done it.